In [1]:
import pandas as pd 

df= pd.read_csv(r'C:\Users\ASUS\OneDrive\Documents\HR-Attrition-Dashboard\SampleData_-HR-Employee-Attrition.csv')

print("Shape:",df.shape)
print("\nNull Values:\n",df.isnull().sum())
print("\nAttrition Count:\n", df['Attrition'].value_counts())

Shape: (1470, 35)

Null Values:
 Age                         0
Attrition                   0
BusinessTravel              0
DailyRate                   0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeNumber              0
EnvironmentSatisfaction     0
Gender                      0
HourlyRate                  0
JobInvolvement              0
JobLevel                    0
JobRole                     0
JobSatisfaction             0
MaritalStatus               0
MonthlyIncome               0
MonthlyRate                 0
NumCompaniesWorked          0
Over18                      0
OverTime                    0
PercentSalaryHike           0
PerformanceRating           0
RelationshipSatisfaction    0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           0
TrainingTimesLastYear       0
WorkLifeBalance             0
YearsAtCompany              0
YearsIn

In [2]:
df= df.drop(columns=['EmployeeCount', 'Over18', 'StandardHours'])
df['AttritionFlag']= df['Attrition'].map({'Yes':1, 'No':0})
df.to_csv('HR_Attrition_Clean.csv',index=False)

In [3]:
print("Clean data shape:", df.shape)
print("\nColumns remaining:", list(df.columns))
print("\nAttritionFlag check:\n", df[['Attrition', 'AttritionFlag']].head(5))

Clean data shape: (1470, 33)

Columns remaining: ['Age', 'Attrition', 'BusinessTravel', 'DailyRate', 'Department', 'DistanceFromHome', 'Education', 'EducationField', 'EmployeeNumber', 'EnvironmentSatisfaction', 'Gender', 'HourlyRate', 'JobInvolvement', 'JobLevel', 'JobRole', 'JobSatisfaction', 'MaritalStatus', 'MonthlyIncome', 'MonthlyRate', 'NumCompaniesWorked', 'OverTime', 'PercentSalaryHike', 'PerformanceRating', 'RelationshipSatisfaction', 'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear', 'WorkLifeBalance', 'YearsAtCompany', 'YearsInCurrentRole', 'YearsSinceLastPromotion', 'YearsWithCurrManager', 'AttritionFlag']

AttritionFlag check:
   Attrition  AttritionFlag
0       Yes              1
1        No              0
2       Yes              1
3        No              0
4        No              0


## Phase 1 - Data Cleaning
- Dataset: IBM HR Analytics (1470 rows, 35 columns)
- No null values found
- Dropped 3 zero-variance columns: EmployeeCount, Over18, StandardHours
- Encoded Attrition column to numeric (Yes=1, No=0)
- Saved clean file as HR_Attrition_Clean.csv
